Core/src/org/sleuthkit/autopsy/casemodule/ImageDSProcessor.java

```diff
@@ -192,6 +198,60 @@ public void run(DataSourceProcessorProgressMonitor progressMonitor, DataSourcePr
         }
         run(deviceId, imagePath, sectorSize, timeZone, ignoreFatOrphanFiles, md5, sha1, sha256, progressMonitor, callback);
     }
+    
+    /**
+     * Adds a data source to the case database using a background task in a
+     * separate thread and the settings provided by the selection and
+     * configuration panel. Files found during ingest will be sent directly to the
+     * IngestStream provided. Returns as soon as the background task is started.
+     * The background task uses a callback object to signal task completion and
+     * return results.
+     *
+     * This method should not be called unless isPanelValid returns true, and 
+     * should only be called for DSPs that support ingest streams.
+     * 
+     * @param progress        Progress monitor that will be used by the
+     *                        background task to report progress.
+     * @param callBack        Callback that will be used by the background task
+     *                        to return results.
+     * @param ingestStream    The ingest stream to send data to
+     */
+    @Override
+    public void run(DataSourceProcessorProgressMonitor progress, DataSourceProcessorCallback callBack, IngestStream ingestStream) {
+        this.ingestStream = ingestStream;
+        if (!setDataSourceOptionsCalled) {
+            configPanel.storeSettings();
+            deviceId = UUID.randomUUID().toString();
+            imagePath = configPanel.getContentPaths();
+            sectorSize = configPanel.getSectorSize();
+            timeZone = configPanel.getTimeZone();
+            ignoreFatOrphanFiles = configPanel.getNoFatOrphans();
+            md5 = configPanel.getMd5();
+            if (md5.isEmpty()) {
+                md5 = null;
+            }
+            sha1 = configPanel.getSha1();
+            if (sha1.isEmpty()) {
+                sha1 = null;
+            }
+            sha256 = configPanel.getSha256();
+            if (sha256.isEmpty()) {
+                sha256 = null;
+            }
+        }
+                
+        run(deviceId, imagePath, sectorSize, timeZone, ignoreFatOrphanFiles, md5, sha1, sha256, progress, callBack);
+    }
```

Comment:
I think we can reduce code duplication using DefaultIngestStream to call run() from run(DataSourceProcessorProgressMonitor progressMonitor, DataSourceProcessorCallback callback).